packages/venia-concept/src/components/Checkout/braintreeDropin.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * @fileoverview This component uses the BrainTree Web Drop In to
+ * hook into Web Payments and the Payment Request API to
+ * submit payments via BrainTree.
+ *
+ * See:
+ * https://github.com/braintree/braintree-web-drop-in
+ * https://braintree.github.io/braintree-web-drop-in/docs/current/index.html
+ * https://developers.braintreepayments.com/guides/drop-in/overview/javascript/v3
+ */
+
+import React, { Component } from 'react';
+import { bool, func } from 'prop-types';
+import dropin from 'braintree-web-drop-in';
+
+const BRAINTREE_SANDBOX_TOKEN = 'sandbox_8yrzsvtm_s2bg8fs563crhqzk';
+
+class BraintreeDropin extends Component {
+    static propTypes = {
+        isRequestingPaymentNonce: bool,
+        onError: func.isRequired,
+        onSuccess: func.isRequired
+    };
+
+    state = {
+        isSupported: true
+    };
+
+    componentDidMount() {
+        this.createDropinInstance()
+            .then(instance => {
+                this.braintree = instance;
+            })
+            .catch(() => {
+                this.setState({ isSupported: false });
+            });
+    }
+
+    componentWillUnmount() {
+        this.braintree = null;
+    }
+
+    componentDidUpdate(prevProps) {
+        if (
+            this.braintree &&
+            this.props.isRequestingPaymentNonce &&
+            !prevProps.isRequestingPaymentNonce
+        ) {
+            // Our parent is telling us to request the payment nonce.
+            this.requestPaymentNonce();
+        }
+    }
+
+    render() {
+        if (!this.state.isSupported) {
+            throw new Error('Web Payments is not supported by this browser.');
+        }
```

Comment:
This is actually still a vestige of when I thought I'd need to provide a fallback UI in case this browser did not support the Payment Request API. Since that is not a concern (see PR write up), I can rip this out.

Reply:
Resolved by new commit.